Added docker-compose directory

surajssd opened the pull request:
added two sub-directories to this directory one for ansible-compose and openshift import

surajssd committed:
Added docker-compose directory

added two sub-directories to this directory one for ansible-compose
and openshift import

rafabene commented on docker-compose-files/openshift-compose/README.md:
````diff
@@ -0,0 +1,45 @@
+Download the docker-compose.yml file.
+
+```bash
+curl -O https://gist.githubusercontent.com/surajssd/336657fc748dcaf595c6e2df5a54d7db/raw/e60f5cadbcd9796be67f71fc1ddb1be91ac3b0b3/docker-compose.yml
````
The URL refers to your repo. It should point to the master branch of this repo.

surajssd replied:
oh yes, thanks for pointing it out!

rafabene commented on docker-compose-files/openshift-compose/README.md:
````diff
@@ -0,0 +1,48 @@
+
+To get latest openshift edit Vagrantfile make `IMAGE_TAG` to `IMAGE_TAG="latest"`
+```bash
+vagrant up && vagrant provision
````
Can this step be made if an existing Vagrant box with OSE 3.1 exists? Or should it be performed before the creation of the box?

surajssd replied:
This step assumes you have Vagrantfile required for running ADB/CDK available with you.

rafabene commented on docker-compose-files/openshift-compose/docker-compose.yml:
```diff
@@ -0,0 +1,143 @@
+    - "hystrix.enabled=true"
+
+  hystrix-dashboard:
+    image: fabric8/hystrix-dashboard:1.0.15
```
Fabric8 Kubeflix has been updated to 1.0.17. Can you try with it?

surajssd replied:
So these labels makes no sense right now, because OpenShift oc import ignores these labels, so updating the PR by removing those labels. 
Issue regarding this has been filed in https://github.com/openshift/origin/issues/9153
And since these labels don't work right now, so Hystix and turbine don't work, I also tried manually labeling services and pods but it didn't help, so plan to drop Hystrix(upon import) was agreed upon.

rafabene commented on docker-compose-files/openshift-compose/docker-compose.yml:
```diff
@@ -0,0 +1,143 @@
+
+
+  turbine-server:
+    image: fabric8/turbine-server:1.0.15
```
Same here

rafabene commented on docker-compose-files/openshift-compose/docker-compose.yml:
```diff
@@ -0,0 +1,143 @@
+    labels:
+    - "project=turbine-server"
+    - "provider=fabric8"
+    - "version=1.0.15"
```
and here

rafabene commented on docker-compose-files/openshift-compose/docker-compose.yml:
```diff
@@ -0,0 +1,143 @@
+    labels:
+    - "project=hystrix-dashboard"
+    - "provider=fabric8"
+    - "version=1.0.15"
```
here

rafabene commented on docker-compose-files/openshift-compose/docker-compose.yml:
```diff
@@ -0,0 +1,143 @@
+    - "8080"
+    labels:
+    - "app=frontend"
+    - "hystrix.enabled:true"
```
You made me realize that the frontend doesn't support hystrix. I just removed it in this commit: https://github.com/redhat-helloworld-msa/helloworld-msa/commit/7e21b725611cb5724f1c2a823f91e93ba0426f6d - You should also remove this label

surajssd replied:
Yes makes sense.

concaf commented:
To try out the application with Ansible:

[docker-compose.yaml](https://gist.githubusercontent.com/containscafeine/64524dcb803b23b7a8d477c0f5b3fc83/raw/a4263cd893f6515715346fb2fbc1fd6211a80ddf/docker-compose.yaml)
[msa_env_export](https://gist.githubusercontent.com/dustymabe/2735d61e8c8f7a79eabbfaa4fdbc96db/raw/c65563c39e3379ffd43575935020f2b45552386a/msa_env_export)
[ansible-compose.yaml](https://gist.githubusercontent.com/containscafeine/32fc5b317e8e53b518b8ddd00e4c1dac/raw/9f109ee3091235f1dc3fb5388cac7cdd79892b4b/ansible-compose.yaml)

concaf committed:
add ansible compose docs and files

rafabene commented on docker-compose-files/ansible-compose/README.md:
````diff
@@ -0,0 +1,99 @@
+Before we get the latest versions of docker and docker-py, let's remove the existing packages on ADB.
+
+```bash
+sudo yum remove -y python-docker-py docker*
````
I believe that this is something that should be executed inside CDK, right? If so, Can you include the vagrant ssh instructions? Thanks

concaf replied:
Done in https://github.com/redhat-helloworld-msa/helloworld-msa/pull/7/commits/7387847844499693d981fc77cb3b8a8d2ee9455d

surajssd committed:
Removed labels from all services

removed labels from all services alongwith hystrix-dashboard service
and turbine server.

concaf committed:
add ADB/CDK info

rafabene commented on docker-compose-files/ansible-compose/README.md:
````diff
@@ -0,0 +1,106 @@
+Bring the application up or down using the following commands:
+
+```bash
+ansible-playbook -t up ansible-compose.yaml
````
I tried it, but the downloaded file is docker-compose.yaml and not ansible-compose.yaml.
When I ran ansible-playbook -t up docker-compose.yaml it replied with:
`[WARNING]: Host file not found: /etc/ansible/hosts

 [WARNING]: provided hosts list is empty, only localhost is available

ERROR! playbooks must be a list of plays

The error appears to have been in '/home/vagrant/msa/docker-compose.yaml': line 1, column 1, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

version: "2"
^ here`

concaf replied:
Hi Rafael,
You can get the ansible-compose.yaml from [here](https://gist.githubusercontent.com/containscafeine/32fc5b317e8e53b518b8ddd00e4c1dac/raw/9f109ee3091235f1dc3fb5388cac7cdd79892b4b/ansible-compose.yaml), I have also updated the same in my comment and in the README.md as well, here - https://github.com/redhat-helloworld-msa/helloworld-msa/pull/7/commits/b7a536d19b3415c7d0699f23555cae78c00f6ebf

rafabene commented on docker-compose-files/ansible-compose/README.md:
````diff
@@ -0,0 +1,106 @@
+```bash
+ansible-playbook -t up ansible-compose.yaml
+ansible-playbook -t down ansible-compose.yaml
+```
````
I believe that a we should also add instructions on how to access the frontend after I run the -t up command.

concaf replied:
Done in https://github.com/redhat-helloworld-msa/helloworld-msa/pull/7/commits/b7a536d19b3415c7d0699f23555cae78c00f6ebf

rafabene commented on docker-compose-files/openshift-compose/README.md:
```diff
@@ -0,0 +1,62 @@
+
+Start ADB/CDK
+
+To get latest openshift edit Vagrantfile make `IMAGE_TAG` to `IMAGE_TAG="latest"`
```
I couldn't find the IMAGE_TAG in the CDK 2.0 Vagrantfile. Should it use another unreleased CDK version?

concaf committed:
add ansible-compose.yaml, update README.md

rafabene commented:
I pushed everything in a squashed commit in a branch called compose: https://github.com/redhat-helloworld-msa/helloworld-msa/tree/compose

So I'm closing this PR. You can send more PRs to that branch. Once we have everything in its place we can merge it to the master branch.
